ftest.sh

```diff
@@ -54,7 +54,7 @@ trap 'echo "encountered an unchecked return code, exiting with error"' ERR
 
 IFS=" " read -r -a nodes <<< "${2//,/ }"
 TEST_NODES=$(IFS=","; echo "${nodes[*]:1:8}")
-
+IFS="," read -r -a CONTROL_NODE <<< "${TEST_NODES}"
```

Comment:
I'm still not understanding why you are parsing `${TEST_NODES}` back into `$CONTROL_NODE` when you can just pick it out of `${nodes[]}`.  `${TEST_NODES}` is already derived from `${nodes}`.  I don't see what advantage there is to ignoring the parsing that has already been done and doing it again.

Reply:
I'm still not convinced of the need, but it's not a major issue.